tests/test_mozillaclub.py

```diff
@@ -0,0 +1,287 @@
+        # in a cache
+        cache = Cache(self.tmp_path)
+        mozillaclub = MozillaClub(MozillaClub_FEED_URL, cache=cache)
+
+        events = [event for event in mozillaclub.fetch()]
+        self.assertEqual(len(http_requests), 1)
+
+        # Now, we get the events from the cache.
+        # The contents should be the same and there won't be
+        # any new request to the server
+        cached_events = [event for event in mozillaclub.fetch_from_cache()]
+        self.assertEqual(len(cached_events), len(events))
+        self.assertEqual(len(http_requests), 1)  # no more requests done
+
+    def test_fetch_from_empty_cache(self):
+        """Test if there are not any events returned when the cache is empty"""
+
+        cache = Cache(self.tmp_path)
+        mozillaclub = MozillaClub(MozillaClub_FEED_URL, cache=cache)
+        cached_events = [event for event in mozillaclub.fetch_from_cache()]
+        self.assertEqual(len(cached_events), 0)
+
+    def test_fetch_from_non_set_cache(self):
+        """Test if a error is raised when the cache was not set"""
+
+        mozillaclub = MozillaClub(MozillaClub_FEED_URL)
+
+        with self.assertRaises(CacheError):
+            _ = [event for event in mozillaclub.fetch_from_cache()]
+
+
+class TestMozillaClubCommand(unittest.TestCase):
+
+    @httpretty.activate
+    def test_parsing_on_init(self):
+        """Test if the class is initialized"""
+
+        args = ['--tag', 'test', MozillaClub_FEED_URL]
+
+        cmd = MozillaClubCommand(*args)
+        self.assertIsInstance(cmd.parsed_args, argparse.Namespace)
+        self.assertEqual(cmd.parsed_args.url, MozillaClub_FEED_URL)
+        self.assertEqual(cmd.parsed_args.tag, 'test')
+        self.assertIsInstance(cmd.backend, MozillaClub)
+
+    def test_argument_parser(self):
+        """Test if it returns a argument parser object"""
+
+        parser = MozillaClubCommand.create_argument_parser()
+        self.assertIsInstance(parser, argparse.ArgumentParser)
+
+
+class TestMozillaClubClient(unittest.TestCase):
+    """MozillaClub API client tests
+
+    These tests not check the body of the response, only if the call
+    was well formed and if a response was obtained. Due to this, take
+    into account that the body returned on each request might not
+    match with the parameters from the request.
+    """
+    @httpretty.activate
+    def test_init(self):
+        """Test initialization"""
+        client = MozillaClubClient(MozillaClub_FEED_URL)
+
+    @httpretty.activate
+    def test_get_events(self):
+        """Test get_events API call"""
+
+        # Set up a mock HTTP server
+        body = read_file('data/mozillaclub/feed.json')
+        httpretty.register_uri(httpretty.GET,
+                               MozillaClub_FEED_URL,
+                               body=body, status=200)
+
+        client = MozillaClubClient(MozillaClub_FEED_URL)
+        response = client.get_cells()
+
+        self.assertEqual(response, body)
+
```

Comment:
I miss here the tests for the parser class. All classes must have their own unit tests.